docs/contributor/arch/011-kymastatsmetrics.md

```diff
@@ -0,0 +1,123 @@
+
+The [kube-state-metrics](https://github.com/kubernetes/kube-state-metrics) exporter provides a way to [export metrics for custom resources](https://github.com/kubernetes/kube-state-metrics/blob/main/docs/metrics/extend/customresourcestate-metrics.md).
+This document describes a way to integrate similar functionality to the Telemetry module's OpenTelemetry Collectors. The solution should avoid the maintenance overhead of an additional third-party-image and allow a dynamic configuration, based on the active Kyma modules.
+
+We investigated the following existing solutions:
+
+- The [Kubernetes Cluster Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/k8sclusterreceiver) is an OpenTelemetry Collector receiver with similar scope as kube-state-metrics . However, this receiver does not support monitoring custom resources.
+- Another available source to monitor changes of arbitrary Kubernetes objects is the [Kubernetes Objects Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/k8sobjectsreceiver). This receiver produces only logs and no metrics.
+
+- The OpenTelemetry Collector provides interfaces for enhancements by implementing custom receiver plugins. The OpenTelemetry project provides [documentation](https://opentelemetry.io/docs/collector/building/receiver/) on implementing a custom receiver and adding it to a custom distribution of the Collector.
+
+## Decision
+
+Due to the restrictions of available telemetry resources for Kubernetes resources, building a custom receiver is the most suitable option.
+
+### Extended MetricPipeline API
+
+To activate module status metrics as a new input, the MetricPipeline CRD needs a _module list_. If the module list is empty, metrics of all active modules are collected. As in the following example, users can select the Kyma modules of interest:
+
+```yaml
+apiVersion: telemetry.kyma-project.io/v1alpha1
+kind: MetricPipeline
+metadata:
+  name: sample
+spec:
+  input:
+    kyma:
+      enabled: true
+      modules:
+        - telemetry
+    runtime:
+      enabled: true
+    prometheus:
+      enabled: true
+    istio:
+      enabled: true
+  output:
+    otlp:
+      endpoint:
+        value: http://example.com:4317
+```
+
+Enabling the Kyma input will enable a custom metrics receiver, called `kymastats`, in the Telemetry metric gateway that produces metrics for the module state and status conditions.
+
+The receiver configuration will follow the shown example:
+
+```yaml
+receivers:
+  kymastats:
+    k8s_cluster:
+        auth_type: serviceAccount
+    collection_interval: 30s
+    api_groups:
+    - operator.kyma-project.io
+```
+
+The receiver needs the following properties:
```

Comment:
I'd put this section (line 69-73) _before_ the example of line 57-67.

Reply:
Done